src/main/java/org/codeforamerica/shiba/output/FileDownLoadController.java

```diff
@@ -75,20 +87,71 @@ public FileDownLoadController(
             @PathVariable String applicationId,
             HttpServletRequest request
     ) {
-        String requestIpHeader = Optional.ofNullable(request.getHeader("X-FORWARDED-FOR")).orElse("");
-        String[] ipAddresses = requestIpHeader.split(",");
-        String requestIp = ipAddresses.length > 1 ? ipAddresses[ipAddresses.length - 2].trim() : "<blank>";
+        String requestIp = createRequestIp(request);
 
         applicationEventPublisher.publishEvent(new DownloadCafEvent(applicationId, requestIp));
         ApplicationFile applicationFile = pdfGenerator.generate(applicationId, CAF, CASEWORKER);
 
         return createResponse(applicationFile);
     }
 
+    @GetMapping("/download-docs/{applicationId}")
+    ResponseEntity<byte[]> downloadDocsWithApplicationId(
+            @PathVariable String applicationId,
+            HttpServletRequest request
+    ) throws IOException {
+        String requestIp = createRequestIp(request);
+
+        // TODO: Change this to a Doc download event
+        applicationEventPublisher.publishEvent(new DownloadCafEvent(applicationId, requestIp));
+
+        Application application = applicationRepository.find(applicationId);
+        List<UploadedDocument> uploadedDocs = application.getApplicationData().getUploadedDocs();
+
+        List<ApplicationFile> applicationFiles = new ArrayList<>();
+        byte[] coverPage = pdfGenerator.generate(application, UPLOADED_DOC, CASEWORKER).getFileBytes();
+        for (int i = 0; i < uploadedDocs.size(); i++) {
+            UploadedDocument uploadedDocument = uploadedDocs.get(i);
+            ApplicationFile fileToSend = pdfGenerator.generateForUploadedDocument(uploadedDocument, i, application, coverPage);
+
+            if (fileToSend.getFileBytes().length > 0) {
+                applicationFiles.add(fileToSend);
+            }
+        }
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
This resource output stream needs to get closed. You should use a try-with-resources like we do here 

https://github.com/codeforamerica/shiba/blob/2168e9075cddebb55a4bd6fb75016c922c4df187/src/main/java/org/codeforamerica/shiba/output/pdf/PdfGenerator.java#L104-L130